homeassistant/components/agent_dvr/alarm_control_panel.py

```diff
@@ -0,0 +1,130 @@
+"""Support for Agent DVR Alarm Control Panels."""
+from homeassistant.components.alarm_control_panel import AlarmControlPanelEntity
+from homeassistant.components.alarm_control_panel.const import (
+    SUPPORT_ALARM_ARM_AWAY,
+    SUPPORT_ALARM_ARM_HOME,
+    SUPPORT_ALARM_ARM_NIGHT,
+)
+from homeassistant.const import (
+    STATE_ALARM_ARMED_AWAY,
+    STATE_ALARM_ARMED_HOME,
+    STATE_ALARM_ARMED_NIGHT,
+    STATE_ALARM_DISARMED,
+)
+
+from .const import CONNECTION, DOMAIN as AGENT_DOMAIN
+
+ICON = "mdi:security"
+
+CONF_HOME_MODE_NAME = "home"
+CONF_AWAY_MODE_NAME = "away"
+CONF_NIGHT_MODE_NAME = "night"
+
+CONST_ALARM_CONTROL_PANEL_NAME = "Alarm Panel"
+
+
+async def async_setup_entry(
+    hass, config_entry, async_add_entities, discovery_info=None
+):
+    """Set up the Agent DVR Alarm Control Panels."""
+    async_add_entities(
+        [AgentBaseStation(hass.data[AGENT_DOMAIN][config_entry.entry_id][CONNECTION])]
+    )
+    return True
+
+
+class AgentBaseStation(AlarmControlPanelEntity):
+    """Representation of an Agent DVR Alarm Control Panel."""
+
+    def __init__(self, client):
+        """Initialize the alarm control panel."""
+        self._state = None
+        self._client = client
+        self._uniqueid = f"{client.unique}_CP"
```

Comment:
Please separate words with underscore in variable names, and rename to `self._unique_id`.